### ISSUE TITLE
Operations in kernel can't be interrupted from ipython notebook

### ISSUE THREAD

**Issue by MichaelSolotky** [bigartm/bigartm#932]
It's inconvinient, I think we should look at the way how binary is called from python and add some flag that it should be able to receive signals.

**Comment by ofrei** [comment]
@MichaelSolotky good point! I'm not familiar with ipython environment - how would you normally interrupt an operation? For example in jupyter notebook, if I have a hanging cell, I know just one way out - simply restart the kernel.  Is there a better way? One typical example  would be pd.read_table, pd.merge or pd.concat - do you know how to properly interrupt these?

**Comment by MichaelSolotky** [comment]
> @MichaelSolotky good point! I'm not familiar with ipython environment - how would you normally interrupt an operation? For example in jupyter notebook, if I have a hanging cell, I know just one way out - simply restart the kernel. Is there a better way? One typical example would be pd.read_table, pd.merge or pd.concat - do you know how to properly interrupt these?

Hm
I didn't try to interrupt pandas operations, but basic python operations can be easily interrupted clicking the "interrupt" button, which actually doesn't work with hard numpy operations. And I don't know what would work better in that situations then restart of the kernel.
I've seen the same behavior with subprocess function in python - I just couldn't stop the execution of a binary file (it's not about bigartm) and there was no jupyter, I was calling it from the command line. So it seems to me it's about the correct way of calling processes from python but not about jupyter.
